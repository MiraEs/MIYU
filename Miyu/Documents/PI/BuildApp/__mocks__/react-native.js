/* eslint-disable react/no-multi-comp */
import {
	createElement,
	Component,
} from 'react';
import PropTypes from 'prop-types';
const ReactNative = jest.genMockFromModule('react-native');

ReactNative.StyleSheet = {
	create: function create(styles) {
		return styles;
	},
	flatten: function flatten(...styles) {
		return Object.assign({}, ...styles);
	},
};

ReactNative.PixelRatio = {
	get() {
		return 1;
	},
};

ReactNative.Alert = {
	alert: jest.fn(),
};

ReactNative.StatusBar = {
	setBarStyle: jest.fn(),
	setHidden: jest.fn(),
};

ReactNative.Linking = {
	openURL: jest.fn(),
};

ReactNative.NativeModules = {
	PhotoUploader: {},
	SafariViewManager: {},
	UIManager: {
		measure: jest.fn(),
	},
	MarketingCloud: {
		trackAction: jest.fn(),
		trackState: jest.fn(),
	},
	LocalyticsManager: {
		tagScreen: jest.fn(),
		tagEvent: jest.fn(),
		tagCustomerLoggedOut: jest.fn(),
	},
	Instabug: {
		log: jest.fn(),
	},
	BNDevice: {},
	RNApplePayManager: {
		canMakePayments: jest.fn(() => Promise.resolve(true)),
		canMakePaymentsUsingNetworks: jest.fn(() => Promise.resolve(true)),
		paymentRequest: jest.fn(),
		paymentRequestDidFinish: jest.fn(),
		updateShippingMethods: jest.fn(),
		selectShippingMethod: jest.fn(),
		authorizedPayment: jest.fn(),
	},
};

ReactNative.requireNativeComponent = jest.fn();

ReactNative.NetInfo = {
	isConnected: {
		fetch: jest.fn(() => ({ done: jest.fn() })),
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
	},
};

ReactNative.AppState = {
	addEventListener: jest.fn(),
};

ReactNative.CameraRoll = {
	getPhotos: jest.fn(() => Promise.resolve([])),
};

ReactNative.PushNotificationIOS = {
	setApplicationIconBadgeNumber: jest.fn(),
};

ReactNative.Keyboard = {
	dismiss: jest.fn(),
};

ReactNative.findNodeHandle = jest.fn();

ReactNative.ViewPropTypes = {
	style: jest.fn(),
};

ReactNative.NativeEventEmitter = class NativeEventEmitter {
	addListener = jest.fn(() => ({ remove: jest.fn() }));
};

class FlatList extends Component {
	render() {
		return createElement('FlatList', this.props);
	}
}

class SectionList extends Component {
	render() {
		return createElement('SectionList', this.props);
	}
}

class View extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('View', this.props, this.props.children);
	}
}

class ScrollView extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('ScrollView', this.props, this.props.children);
	}
}

class Switch extends Component {
	render() {
		return createElement('Switch');
	}
}

class Text extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Text', this.props, this.props.children);
	}
}

class TextInput extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TextInput', this.props, this.props.children);
	}
}

class TouchableOpacity extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableOpacity', this.props, this.props.children);
	}
}

class TouchableWithoutFeedback extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableWithoutFeedback', this.props, this.props.children);
	}
}

class ToolbarAndroid extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ToolbarAndroid', this.props, this.props.children);
	}
}

class Image extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Image', this.props, this.props.children);
	}
}

class Icon extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('Icon', this.props, this.props.children);
	}
}

class ActivityIndicator extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ActivityIndicator', this.props, this.props.children);
	}
}

ReactNative.Platform = {
	OS: 'ios',
	select: jest.fn((options) => options.ios),
};

ReactNative.InteractionManager = {
	runAfterInteractions: jest.fn(),
};

ReactNative.ActionSheetIOS = {
	showActionSheetWithOptions: jest.fn(),
};

class ListView extends Component {
	DataSource() {
		return {
			cloneWithRows: jest.fn(),
			cloneWithRowsAndSections: jest.fn(),
		};
	}
}

class AppRegistry {
	registerComponent() {}
}

class TouchableHighlight extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableHighlight', this.props, this.props.children);
	}
}
class BackHandler extends Component {
	static addEventListener() {}
	static exitApp() {}
	static removeEventListener() {}
}

class TouchableNativeFeedback extends Component {
	static Ripple() {
	}
}

class Navigator extends Component {
	render() {
	}
}

class RefreshControl extends Component {
	render() {
		return createElement('RefreshControl', this.props);
	}
}

class LayoutAnimation extends Component {
	static configureNext() {
	};

	static Presets() {
		return {
			easeInEaseOut: jest.fn(),
		};
	};
}

class Dimensions {
	static get() {
		return 0;
	}
}

class Animated extends Component {

	static Value() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static ValueXY() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static event() {}
}

class AnimatedView extends Component {
	render() {
		return createElement('AnimatedView');
	}
}

class AnimatedImage extends Component {

}

class AnimatedText extends Component {

}

Animated.View = AnimatedView;

Animated.Image = AnimatedImage;

Animated.Text = AnimatedText;

class PanResponder {
	static create() {
		return {panHandlers: {}};
	}
}

global.__fbBatchedBridgeConfig = {
	remoteModuleConfig: [],
	localModulesConfig: [],
};

ReactNative.View = View;
ReactNative.ScrollView = ScrollView;
ReactNative.FlatList = FlatList;
ReactNative.ListView = ListView;
ReactNative.Text = Text;
ReactNative.TextInput = TextInput;
ReactNative.TouchableOpacity = TouchableOpacity;
ReactNative.TouchableHighlight = TouchableHighlight;
ReactNative.TouchableWithoutFeedback = TouchableWithoutFeedback;
ReactNative.ToolbarAndroid = ToolbarAndroid;
ReactNative.Image = Image;
ReactNative.AppRegistry = AppRegistry;
ReactNative.Icon = Icon;
ReactNative.ActivityIndicator = ActivityIndicator;
ReactNative.BackHandler = BackHandler;
ReactNative.TouchableNativeFeedback = TouchableNativeFeedback;
ReactNative.Navigator = Navigator;
ReactNative.RefreshControl = RefreshControl;
ReactNative.LayoutAnimation = LayoutAnimation;
ReactNative.Dimensions = Dimensions;
ReactNative.Animated = Animated;
ReactNative.PanResponder = PanResponder;
ReactNative.SectionList = SectionList;
ReactNative.Switch = Switch;

module.exports = ReactNative;
