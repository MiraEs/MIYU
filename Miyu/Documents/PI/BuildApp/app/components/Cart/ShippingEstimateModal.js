import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	View,
	StyleSheet,
	TouchableOpacity,
} from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
	Text,
	Button,
} from 'BuildLibrary';
import styles from '../../lib/styles';
import helpers from '../../lib/helpers';
import EventEmitter from '../../lib/eventEmitter';
import Form from '../Form';
import FormInput from '../FormInput';
import LoadingView from '../LoadingView';
import { FREE_SHIPPING } from '../../constants/constants';
import TrackingActions from '../../lib/analytics/TrackingActions';
import {
	updateSessionCart,
	setSelectedShippingIndex,
} from '../../actions/CartActions';
import { showAlert } from '../../actions/AlertActions';
import PhoneHelper from '../../lib/PhoneHelper';

const componentStyles = StyleSheet.create({
	screen: {
		flex: 1,
		justifyContent: 'flex-start',
	},
	contentWithInput: {
		backgroundColor: styles.colors.white,
		paddingHorizontal: styles.measurements.gridSpace1,
		borderBottomWidth: styles.dimensions.borderWidth,
		borderColor: styles.colors.grey,
	},
	contentWithList: {
		flex: 1,
		backgroundColor: styles.colors.greyLight,
		paddingHorizontal: styles.measurements.gridSpace1,
		paddingTop: styles.measurements.gridSpace2,
		paddingBottom: styles.measurements.gridSpace1,
	},
	border: {
		padding: styles.dimensions.borderWidth,
		borderWidth: styles.dimensions.borderWidth,
		borderColor: styles.colors.grey,
	},
	borderSelected: {
		borderWidth: styles.dimensions.borderWidthLarge,
		borderColor: styles.colors.accent,
	},
	shippingOption: {
		flexDirection: 'row',
		alignItems: 'center',
		backgroundColor: styles.colors.white,
		marginBottom: styles.measurements.gridSpace1,
		paddingHorizontal: styles.measurements.gridSpace1,
		paddingTop: styles.measurements.gridSpace1,
		paddingBottom: styles.measurements.gridSpace2,
	},
	shippingCost: {
		flexDirection: 'row',
	},
});

export class ShippingEstimateModal extends Component {

	constructor(props) {
		super(props);

		this.state = {
			isLoading: false,
			selectedShippingIndex: props.selectedShippingIndex,
			shippingOptions: props.cart.shippingOptions || [],
			zipCode: props.cart.zipCode,
		};
	}

	componentWillReceiveProps(nextProps) {
		const { cart } = nextProps;

		this.setState({
			shippingOptions: cart.shippingOptions || [],
			zipCode: cart.zipCode,
		});
	}

	setScreenTrackingInformation() {
		return {
			name: 'build:app:shippingestimatemodal',
		};
	}

	onChange = (shippingForm) => {
		const zipCode = shippingForm['zipCode'].value || null;

		this.setState({
			noResults: false,
			zipCode,
		}, () => {
			if (!zipCode) {
				this.onEnterZipcode();
			}
		});
	};

	onEnterZipcode = () => {
		const { actions: { showAlert, updateSessionCart }, cart } = this.props;
		const { zipCode } = this.state;

		this.setState({
			isLoading: true,
			noResults: false,
			selectedShippingIndex: -1,
		}, () => {
			updateSessionCart({
				sessionCartId: cart.sessionCartId,
				cart: { zipCode },
			})
			.then(() => {
				this.setState((prevState) => {
					return {
						noResults: prevState.shippingOptions.length === 0,
						isLoading: false,
					};
				});
			})
			.catch(() => {
				showAlert('Invalid ZIP Code', 'error');
				this.setState((prevState) => {
					return {
						zipCode: cart.zipCode,
						noResults: prevState.shippingOptions.length === 0,
						isLoading: false,
					};
				});
			})
			.done();
		});
	};

	onSelectShippingOption = (selectedShippingIndex) => {
		const { actions: { setSelectedShippingIndex }, onClose } = this.props;

		this.setState({ selectedShippingIndex }, () => {
			setSelectedShippingIndex(selectedShippingIndex);
			onClose();
		});
	};

	renderShippingOptions = () => {
		const { shippingOptions } = this.state;
		const { onClose } = this.props;
		const filtered = (shippingOptions) ? shippingOptions.filter((option) => option.customerAvailable) : [];

		if (this.state.isLoading) {
			return <LoadingView backgroundColor={styles.colors.greyLight}/>;
		}

		if (this.state.noResults) {
			return <Text>ZIP Code not found.</Text>;
		}

		if (shippingOptions.length > 0 !== filtered.length > 0) {
			const phone = PhoneHelper.getPhoneNumberByUserType(this.props.user);
			return (
				<View>
					<Text>
						Please contact us at{' '}
						<Text
							color="primary"
							onPress={() => {
								EventEmitter.emit('onCallUs', phone, true);
								onClose();
							}}
						>
							{PhoneHelper.formatPhoneNumber(phone)}
						</Text>
						{' '}for a shipping estimate.{'\n'}
					</Text>
					<Button
						onPress={this.props.onClose}
						text="Back to Cart"
						accessibilityLabel="Back to Cart Button"
						trackAction={TrackingActions.BACK_TO_CART}
					/>
				</View>
			);
		} else {
			return shippingOptions.map((option, index) => {
				const border = index === this.state.selectedShippingIndex ? componentStyles.borderSelected : componentStyles.border;
				let cost = (
					<Text
						color="primaryDark"
						weight="bold"
					>
						{helpers.toUSD(option.shippingCost)}
					</Text>);

				if (option.shippingCost === 0) {
					cost = (
						<View style={componentStyles.shippingCost}>
							<Text decoration="line-through">{helpers.toUSD(FREE_SHIPPING)} </Text>
							<Text
								color="accent"
								weight="bold"
							>
								FREE
							</Text>
						</View>
					);
				}

				return (
					<TouchableOpacity
						key={index}
						onPress={() => this.onSelectShippingOption(index)}
						accessibilityLabel={option.shippingName}
					>
						<View style={[componentStyles.shippingOption, border]}>
							<View>
								<Text>{option.shippingName}</Text>
								{cost}
							</View>
						</View>
					</TouchableOpacity>
				);
			});
		}
	};

	render() {
		return (
			<View style={componentStyles.screen}>
				<View style={componentStyles.contentWithInput}>
					<Form
						ref={(ref) => this.shippingForm = ref}
						onChange={this.onChange}
					>
						<FormInput
							autoCapitalize="characters"
							autoCorrect={false}
							name="zipCode"
							clearButtonMode="always"
							label="ZIP Code:"
							inlineLabel={true}
							onSubmitEditing={this.onEnterZipcode}
							placeholder="Enter Shipping ZIP Code"
							returnKeyType="go"
							value={this.state.zipCode}
							scrollToOnFocus={false}
							keyboardType="numbers-and-punctuation"
							accessibilityLabel="Zip Code Input"
						/>
					</Form>
				</View>
				<View style={componentStyles.contentWithList}>
					{this.renderShippingOptions()}
				</View>
			</View>
		);
	}
}

ShippingEstimateModal.propTypes = {
	actions: PropTypes.object.isRequired,
	cart: PropTypes.object.isRequired,
	selectedShippingIndex: PropTypes.number.isRequired,
	onClose: PropTypes.func.isRequired,
	user: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => {
	return {
		cart: state.cartReducer.cart,
		selectedShippingIndex: state.cartReducer.selectedShippingIndex,
		user: state.userReducer.user,
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		actions: bindActionCreators({
			setSelectedShippingIndex,
			showAlert,
			updateSessionCart,
		}, dispatch),
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(ShippingEstimateModal);
