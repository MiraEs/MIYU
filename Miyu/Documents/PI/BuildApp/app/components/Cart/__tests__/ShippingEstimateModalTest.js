import { create } from 'react-test-renderer';
import React from 'react';
import 'react-native';
import { ShippingEstimateModal } from '../ShippingEstimateModal';

const props = {
	actions: {},
	cart: {},
	selectedShippingIndex: 2,
	onClose: jest.fn(),
	user: {},
};

function setup(otherProps) {
	const wrapper = create(
		<ShippingEstimateModal
			{...props}
			{...otherProps}
		/>
	);
	const instance = wrapper.getInstance();
	return {
		wrapper,
		instance,
	};
}

describe('ShippingEstimateModal', () => {
	it('should render with full props', () => {
		const { wrapper } = setup();
		expect(wrapper.toJSON()).toMatchSnapshot();
	});
});
