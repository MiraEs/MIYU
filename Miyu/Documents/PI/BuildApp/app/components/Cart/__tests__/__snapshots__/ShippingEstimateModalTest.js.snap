// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingEstimateModal should render with full props 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderBottomWidth": 1,
        "borderColor": "#D0D3D4",
        "paddingHorizontal": 8,
      }
    }
  >
    <ScrollView
      accessibilityLabel={undefined}
      keyboardShouldPersistTaps="always"
      onScroll={[Function]}
      scrollsToTop={undefined}
      style={undefined}
    >
      <View
        style={
          Array [
            undefined,
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 8,
            }
          }
          weight="bold"
        >
          ZIP Code:
        </Text>
        <View
          style={
            Object {
              "width": 8,
            }
          }
        />
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#D0D3D4",
                  "borderWidth": 1,
                  "marginVertical": 8,
                  "paddingHorizontal": 16,
                },
                "",
                "",
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                Object {},
              ]
            }
          >
            <TextInput
              accessibilityLabel="Zip Code Input"
              accessible={true}
              autoCapitalize="characters"
              autoCorrect={false}
              clearButtonMode="always"
              hideErrorText={false}
              inlineLabel={true}
              isRequired={false}
              isRequiredError="This field is required."
              keyboardType="numbers-and-punctuation"
              label="ZIP Code:"
              lines={1}
              multiline={false}
              name="zipCode"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Enter Shipping ZIP Code"
              returnKeyType="go"
              scrollToOnAndroid={false}
              scrollToOnFocus={false}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "color": "#63666A",
                    "fontFamily": "ProximaNova-Regular",
                    "fontSize": 16,
                    "lineHeight": 21,
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                  Object {
                    "height": 44,
                  },
                  undefined,
                  "",
                ]
              }
              textInputContainerStyle={Object {}}
              underlineColorAndroid="transparent"
              validateOnBlur={true}
              validateOnChange={false}
              value={undefined}
            />
          </View>
        </View>
      </View>
    </ScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#EBEDEF",
        "flex": 1,
        "paddingBottom": 8,
        "paddingHorizontal": 8,
        "paddingTop": 16,
      }
    }
  />
</View>
`;
